opm/simulators/wells/WellAssemble.cpp

```diff
@@ -161,7 +162,15 @@ assembleControlEqProd(const WellState& well_state,
                 well_.rateConverter().calcCoeff(id, region, coeff);
 
         };
-        WellGroupControls(well_).getGroupProductionControl(group, well_state,
+      
+        bool has_choke(false);
+        const std::size_t report_step = well_.currentStep();
```

Comment:
As mentioned in the private message. We might not want to handle the control equation assembly here.  We might want to treat the well under choke pressure control as a normal THP control.

And instead, when the group node is a choke (the group that gives the rate target), we might not have the GRUP control from that group node. 

Some work when specifying the control types might need to be done. (not sure whether from the parser side yet. )

Reply:
As you observed the `WMCTL`  (Mode of Control) was not in agreement with the reference simulation. I  changed WellInterface_impl.hpp so that a well under `GRUP`control and with auto-choke its `production_cmode` is set to `Well::ProducerCMode::THP` and the handling of the control equation  for this case (GRUP control and with auto-choke) removed from  WellAssemble.cpp as it is not needed anymore as the well is now treated with `THP` control.

The figure below sows that B-1H is under THP control (`WMCTL=6`)
![image](https://github.com/OPM/opm-simulators/assets/56635911/2232c3ed-8a3b-48b2-bcb6-8ab188deb5f3)